Provider result caching

maartenpaauw opened the pull request:
It's unlikely that the result will change within (for example) a day.

maartenpaauw committed:
Add provider result caching

devfrey commented on src/Cache/NullCache.php:
```diff
@@ -0,0 +1,126 @@
+     * @return mixed The value of the item from the cache, or $default in case of cache miss.
+     * @throws \Psr\SimpleCache\InvalidArgumentException
+     *   MUST be thrown if the $key string is not a legal value.
+     */public function get($key, $default = null)
```
Function declaration should start on a new line

devfrey replied:
Function declaration must start on a new line.

devfrey commented on src/Cache/NullCache.php:
```diff
@@ -0,0 +1,126 @@
+
+use Psr\SimpleCache\CacheInterface;
+
+class NullCache implements CacheInterface
```
Perhaps add a description to describe what this class is for.

devfrey replied:
Perhaps add a description to describe what this class is for.

devfrey commented on src/Cache/NullCache.php:
```diff
@@ -0,0 +1,126 @@
+     */
+    public function set($key, $value, $ttl = null)
+    {
+        return false;
```
False indicates an error. I think it'd be better to have all functions simulate a happy flow. So make this null-cache pretend everything's working. The only exception here is the `has()` method.

devfrey commented on tests/Cache/NullCacheTest.php:
```diff
@@ -0,0 +1,66 @@
+use DevMob\Postcodes\Cache\NullCache;
+use DevMob\Postcodes\Tests\TestCase;
+
+class NullCacheTest extends TestCase
```
Be sure to add test cases for all methods (full coverage).

devfrey commented on src/Address.php:
```diff
@@ -0,0 +1,55 @@
+use DevMob\Postcodes\Providers\ProviderInterface;
+use Psr\SimpleCache\CacheInterface;
+
+class Address implements ProviderInterface
```
You should name this core wrapper the same as the package: `Postcodes`.
Usage will be:

```php
$provider = new PostcodeApiNu(['key' => '...']);
$postcodes = new Postcodes($provider);
$results = $postcodes->lookup('3011ED', '50');
```

devfrey commented on src/Address.php:
```diff
@@ -0,0 +1,55 @@
+     * @throws \DevMob\Postcodes\Exceptions\PostcodesException
+     * @throws \Psr\SimpleCache\InvalidArgumentException
+     */
+    public function lookup(string $postcode, string $number): array
```
Add a test for this method.

maartenpaauw committed:
Use the happy flow

False indicates an error. I think it'd be better to have all functions simulate a happy flow. So make this null-cache pretend everything's working.

maartenpaauw committed:
Rename to Postcodes

The core wrapper should be the same as the package.

maartenpaauw committed:
Test all methods
